R/compile_model.R

```diff
@@ -0,0 +1,95 @@
+#' This function reads in and optionally compiles a Stan model.
+#' It is written to search the installed package `stan` directory
+#' for additional source files to include. Within each stan file,
+#' use #include {path to your file with the `stan` directory}.stan
+#'
+#' @details The code for this function has been adapted
+#' from code written (under an MIT license) as part of
+#' the [`epinowcast`](https://github.com/epinowcast/epinowcast)
+#' R package.
+#'
+#' @param model_filepath path to .stan file defining the model, default is
+#' `system.file("stan", "wwinference.stan", package = "wwinference")
+#' @param include_paths path(s) to directories to search for files
+#' specified in #include statements. Passed to [cmdstanr::cmdstan_model()].
+#' Defaults to the `stan` subdirectory of the installed
+#' `wwinference` package.
+#' @param threads Number of threads to use in model compilation,
+#' as an integer. Passed to [cmdstanr::cmdstan_model()].
+#' Default `FALSE` (use single-threaded compilation).
+#' @param target_dir Directory in which to save the compiled
+#' stan model binary. Passed as the `dir` keyword argument to
+#' [cmdstanr::cmdstan_model()]. Defaults to a temporary directory
+#' for the R session (the output of [tempdir()]).
+#' @param stanc_options Options for the stan compiler passed to
+#' [cmdstanr::cmdstan_model()], as a list. See that function's
+#' documentation for more details. Default `list()` (use default
+#' options).
+#' @param cpp_options Options for the C++ compiler passed to
+#' [cmdstanr::cmdstan_model()], as a list. See that function's
+#' documentation for more details. Default `list()` (use default
+#' options).
+#' @param verbose Write detailed output to the terminal while
+#' executing the function? Boolean, default `TRUE`.
+#' @param ... Additional keyword arguments passed to
+#' [cmdstanr::cmdstan_model()].
+#'
+#' @return The resulting `cmdstanr` model object, as the output
+#' of [cmdstanr::cmdstan_model()].
+#' @export
+compile_model <- function(model_filepath = system.file("stan",
+                            "wwinference.stan",
+                            package = "wwinference"
+                          ),
+                          include_paths = system.file(
+                            "stan",
+                            package = "wwinference"
+                          ),
+                          threads = FALSE,
+                          target_dir = tempdir(),
+                          stanc_options = list(),
+                          cpp_options = list(),
+                          verbose = TRUE,
+                          ...) {
+  if (verbose) {
+    cli::cli_inform(
+      glue::glue(paste0(
+        "Using model source file: ",
+        "{model_filepath}"
+      ))
+    )
+    cli::cli_inform(
+      sprintf(
+        "Using include paths: %s",
+        toString(include_paths)
+      )
+    )
+  }
+
+  create_dir(target_dir)
+
+  model <- cmdstanr::cmdstan_model(
```

Comment:
Ah, I had been pretty much using pkg::fn() throughout because I thought it was the preferred syntax.

I realize now it is redundant as long as you don't have duplicate package names (which you sometimes do). I can make a separate issue to go through and replace any instances of `pkg::fun()` with `fun` and add `importfrom` to the `wwinference-package.R` ?

Reply:
https://github.com/CDCgov/ww-inference-model/issues/13

Reply:
Hmm ok we might need to edit our pre-commit then, because removing it results in it failing our dependency check:
```
Not all packages used in your code are listed in DESCRIPTION. The following are missing in file `/home/kej38/github_repos/ww-inference-model/data-raw/covid_pmfs.R`: usethis.
You can add them with `usethis::use_package()` or `usethis::use_dev_package()`.
Error: Dependency check failed
```

Reply:
Clarification: in general I favor `pkg::function_used_sometimes()` and no `importFrom(pkg, function_used_sometimes)`

But *if* there's a `function_used_very_often` in `pkg` *and* for that reason you decide to do `importFrom(pkg, function_used_very_often)`, then when you call it there's no need to write `pkg:: function_used_very_often()` instead of just `function_used_very_often()`.

And yes, definitely don't create clashes with `importFrom`; never do `importFrom(pkg1, shared_name_fn)` and `importFrom(pkg2, shared_name_fn)`

Reply:
Cool, will go ahead and clean this up in the new issue.